### DISCUSSION TITLE
Deployment: 504 Gateway Time-out

### DISCUSSION THREAD

**Discussion by Anaxarchos** [pimcore/pimcore#13250] [Q&A / Support]
Thanks to the help of @BlackbitNeueMedien and @jdreesen I am now able to deploy the code from my local machine to my test server. But from time to time after running `composer update` and bin/console `pimcore:deploy:classes-rebuild --create-classes` and `bin/console pimcore:deploy:cach:clear` I cannot access the website and nginx returns a 504 Error.
I tried restarting nginx, but that did not help. Only after rebooting the server, everything is working again.
I have no corresponding error messages neither in the dev.log nor in the php.log file.
Maybe this is due to my nginx configuration which is much simpler than the one from the pimcore documentation.
Any help is very much appreciated. Thank you.

**Comment by Anaxarchos** [comment]
Maybe it is helpful to extend the max execution time, which I now set to 300. Let's see if the time out error will still occur.

**Reply by Anaxarchos** [reply]
The error still occured and I had a look in the error.log from nginx which told me:
upstream timed out (110: Connection timed out) while reading response header from upstream
Now I added this to my nginx configuration:

```
upstream php-pimcore10 {
        server unix:/var/run/php/php8.0-fpm.sock;
}
```

and that to www.conf:

```
pm = dynamic
pm.max_children = 120
pm.start_servers = 12
pm.min_spare_servers = 6
pm.max_spare_servers = 18

```

**Comment by Anaxarchos** [comment]
Now a 502 bad gateway error occurs from tim to time (once or twice a day), no matter if I include php-fpm with the upstream or with a location directive. The reason for that is that the php-fpm stops and I have to restart it again. When I clear all caches, pimcore kills the php-fpm every minute until the cache is filled again.

There should be a way to configure php.ini, www.conf and nginx in a way that pimcore does not kill the server when rebuilding the cache or thumbnails or whatever. How do you guy do this? 
Thanx

**Reply by jaisdominik** [reply]
Hey, did you figure it out by now? We have the same issue. After cache clear from time to time PHP FPM is killed until the cache is filled again.

Would be great to get some support here

**Reply by Anaxarchos** [reply]
The way Pimcore renders the thumbnails can kill small servers. So I ended up switching to a server with more cores and more ram.

**Reply by dpfaffenbauer** [reply]
@Anaxarchos It's not the way Pimcore renders the thumbnail. It is the way how images are processed in general ;) We have similar "issues" with that on resource limited pods in kubernetes.

**Reply by Anaxarchos** [reply]
@dpfaffenbauer You are right.
